sickbeard/search.py

```diff
@@ -565,7 +565,10 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
 
         # Update the cache if a manual search is being runned
         if manual_snatch:
-            results = curProvider.cache.update_cache_manual_search(searchResults[curEp])
+            results = False
+            for searched_episode in episodes:
+                results = curProvider.cache.update_cache_manual_search(searchResults[searched_episode.episode])
+
             if results:
                 # If we have at least a result from one provider, it's good enough to be marked as result
                 finalResults.append(results)
```

Comment:
The existing code only appends the result for the last found.
With the change of results to a list comprehension this should become a list concat.  In addition the concat should be moved outside the `if` statement which will only be used to continue.